File: 인범/2026_05_07/grabbing_item.py
def solution(rectangle, characterX, characterY, itemX, itemY):
    from collections import deque
    maxi = 0
    for rect in rectangle:
        maxi = max(maxi,max(rect))
    mymap = [[0]*(maxi*2 +1) for _ in range(maxi*2+1)]
    visited = [[False]*(maxi*2 +1) for _ in range(maxi*2+1)]
    
    for rect in rectangle:
        lx,ly,rx,ry = rect
        for i in range(lx*2,rx*2+1):
            for j in range(ly*2,ry*2+1):
                mymap[i][j] = 1
    
    for rect in rectangle:
        lx,ly,rx,ry = rect
        for i in range(lx*2 +1, rx*2):
            for j in range(ly*2 +1, ry*2):
                mymap[i][j] = 0
    
    queue = deque([(characterX*2,characterY*2,0)])
    visited[characterX*2][characterY*2] = True
    direction = [[1,0],[-1,0],[0,1],[0,-1]]
    
    while queue:
        curr_x, curr_y, count = queue.popleft()
        
        if curr_x==itemX*2 and curr_y==itemY*2 : return count//2
        
        for dx,dy in direction:
            nx= curr_x+dx
            ny= curr_y+dy
            
            if 0<=nx<=maxi*2 and 0<=ny<=maxi*2 and mymap[nx][ny]==1 and not visited[nx][ny]:
                queue.append((nx,ny,count+1))
                visited[nx][ny]=True
        
    
    
    
    return -1

File: 인범/2026_05_07/grabbing_item.java
import java.util.*;

class Solution {
    public int solution(int[][] rectangle, int characterX, int characterY, int itemX, int itemY) {
        int answer = 0;
        int[][] newmap = new int[102][102];
        boolean[][] visited = new boolean[102][102];
        int[][] newrect = new int[rectangle.length][4];
        int charX=characterX*2;
        int charY=characterY*2;
        int ix=itemX*2;
        int iy=itemY*2;
        
        for(int i=0; i<rectangle.length; i++){
            newrect[i][0] = rectangle[i][0]*2;
            newrect[i][1] = rectangle[i][1]*2;
            newrect[i][2] = rectangle[i][2]*2;
            newrect[i][3] = rectangle[i][3]*2;
        }
        
        for(int[] rect : newrect){
            int lx=rect[0];
            int ly=rect[1];
            int rx=rect[2];
            int ry=rect[3];
            
            for(int i=lx; i<=rx; i++){
                for(int j=ly; j<=ry; j++){
                    newmap[i][j] = 1;
                }
            }
        }
        for(int[] rect : newrect){
            int lx=rect[0];
            int ly=rect[1];
            int rx=rect[2];
            int ry=rect[3];
        
            for(int i=lx+1; i<rx; i++){
                for(int j=ly+1; j<ry; j++){
                    newmap[i][j] = 0;
                }
            }
        }
        
        int[][] direction = {{1,0},{-1,0},{0,1},{0,-1}};
        Deque<int[]> q = new ArrayDeque<>();
        q.addLast(new int[]{charX,charY,0});
        
        while(!q.isEmpty()){
            int[] curr = q.pollFirst();
            int row = curr[0];
            int col = curr[1];
            int count = curr[2];
            
            if(row==ix && col==iy){return count/2;}
            
            for(int[] dir : direction){
                int nr = row+dir[0];
                int nc = col+dir[1];
                
                if(nr<0 || nr>=102 || nc<0 || nc>=102){continue;}
                if(visited[nr][nc]){continue;}
                if(newmap[nr][nc] == 1){
                    q.addLast(new int[]{nr,nc,count+1});
                    visited[nr][nc] = true;
                }
            }
        }
        
        return -1;
    }
}